test/integration/dialects/postgres/dao.test.js

```diff
@@ -363,38 +363,37 @@ if (dialect.match(/^postgres/)) {
         });
       });
 
-      it('should be able to add enum types', function(done) {
+      it('should be able to add enum types', function() {
         var self = this
           , User = this.sequelize.define('UserEnums', {
               mood: DataTypes.ENUM('happy', 'sad', 'meh')
-            });
-
-        var _done = _.after(4, function() {
-          done();
-        });
+            })
+          , count = 0;
 
-        User.sync({ force: true }).then(function() {
+        return User.sync({ force: true }).then(function() {
           User = self.sequelize.define('UserEnums', {
             mood: DataTypes.ENUM('neutral', 'happy', 'sad', 'ecstatic', 'meh', 'joyful')
           });
 
-          User.sync().then(function() {
+          return User.sync().then(function() {
             expect(User.rawAttributes.mood.values).to.deep.equal(['neutral', 'happy', 'sad', 'ecstatic', 'meh', 'joyful']);
-            _done();
+            count++;
           }).on('sql', function(sql) {
```

Comment:
Needs to be refactored to use `logging` since sql events will no longer exist when we kill the event emitter.